File: LeetCode/Java/Largest3SameDigitNumberInString.java
public class Largest3SameDigitNumberInString {
    public static void main(String[] args) {
        String s = "6777133339";
        
        for (int i = 9; i >= 0; i--) {
            String check = String.valueOf(i).repeat(3); // Java 11+
            if (s.contains(check)) {
                System.out.println(check);
                return;
            }
        }
        
        System.out.println(""); 
    }
}

// Input: num = "6777133339"
// Output: "777"

// Input: num = "2300019"
// Output: "000"

// Input: num = "42352338"
// Output: ""

File: LeetCode/Python/Largest3SameDigitNumberInString.py
def largestGoodInteger(num: str) -> str:
        for i in range(9,-1,-1):
            check=str(i)*3
            if check in num:
                return check
        return ""
    
s="6777133339"
print(largestGoodInteger(s))

# Input: num = "6777133339"
# Output: "777"

# Input: num = "2300019"
# Output: "000"

# Input: num = "42352338"
# Output: ""